app/views/editor/level/thangs/AddThangsView.coffee

```diff
@@ -23,8 +25,23 @@ module.exports = class AddThangsView extends CocoView
     super options
     @world = options.world
 
+    @thangTypes = new Backbone.Collection()
+    thangTypeCollection = new ThangTypeSearchCollection([])
+    thangTypeCollection.fetch({data: {limit: PAGE_SIZE}})
+    thangTypeCollection.skip = 0
     # should load depended-on Components, too
-    @thangTypes = @supermodel.loadCollection(new ThangTypeSearchCollection(), 'thangs').model
+    @supermodel.loadCollection(thangTypeCollection, 'thangs')
+    @listenTo thangTypeCollection, 'sync', @onThangCollectionSynced
+
+  onThangCollectionSynced: (collection) ->
+    getMore = collection.models.length is PAGE_SIZE
```

Comment:
This will only ever skip once. Going from 1000 to 2000. Would you want to keep iterating until you have all of the thangs?

Reply:
Yes, for example if we reduce 1000 limit later or will have more than 2000 thangs. The same mthod is used for i18n home page

Reply:
I guess what I'm wondering is. You only skip more if you have the `PAGE_SIZE` amount. So if you `PAGE_SIZE` is 1000. Then you will have 1000 thangs after the first iteration. `getMore` is now true. But if you now have 2000 thangs, `getMore` will now be false even if there are more thangs to get. Maybe there are 2001 thangs?
Should the condition be when you getMore and the size didn't increase then stop fetching more thangs?

Reply:
`collection` is not an aggregated collection here and contains only models which were fetched. That's why we add them in the aggregated collection `thangTypes` then. On the first iteration `collection` has 1000 elements, on the second 28 (checked)

Reply:
Thank you Bryukh for explaining it to me :)
Your code looks great.